### PR TITLE
feat(build): add git version to binary and huge pages check

### PR DESCRIPTION
**Pull request by gila**
Some minor changes:

 - add check for huge pages or refuse to start
 - add git_version to binary
 - fix some clippies

### CODE REVIEW

**Comment by gila** [comment]
bors try

**Review by paulyoong — APPROVED** [review]

File: mayastor/src/bin/mayastor.rs [hunk 1]
```diff
@@ -60,7 +63,30 @@ fn start_tokio_runtime(args: &MayastorCliArgs) {
...
-const VERSION_STR: &str = "0.9.0 C2.RC2";
+fn hugepage_check() {
+    let hugepage_path = Path::new("/sys/kernel/mm/hugepages/hugepages-2048kB");
+    let nr_pages: u32 = sysfs::parse_value(&hugepage_path, "nr_hugepages")
+        .expect("failed to read the number of pages");
+    let free_pages: u32 = sysfs::parse_value(&hugepage_path, "free_hugepages")
+        .expect("failed to read the number of free pages");
+    if nr_pages < PAGES_NEEDED {
+        error!(
+            "insufficient huge pages allocated, needed:  {}  allocated: {}",
```
Comment (jonathan-teh) [inline mayastor/src/bin/mayastor.rs:75]: Nit: extra space after the first colon, needs comma after first close curly bracket.

File: mayastor/src/bin/mayastor.rs [hunk 2]
```diff
@@ -60,7 +63,30 @@ fn start_tokio_runtime(args: &MayastorCliArgs) {
...
 }
 
-const VERSION_STR: &str = "0.9.0 C2.RC2";
+fn hugepage_check() {
+    let hugepage_path = Path::new("/sys/kernel/mm/hugepages/hugepages-2048kB");
+    let nr_pages: u32 = sysfs::parse_value(&hugepage_path, "nr_hugepages")
+        .expect("failed to read the number of pages");
+    let free_pages: u32 = sysfs::parse_value(&hugepage_path, "free_hugepages")
+        .expect("failed to read the number of free pages");
+    if nr_pages < PAGES_NEEDED {
```
Comment (tiagolobocastro) [inline mayastor/src/bin/mayastor.rs:73]: Should this number vary depending on the number of configured cores?

File: mayastor/src/bin/mayastor.rs [hunk 3]
```diff
@@ -60,7 +63,30 @@ fn start_tokio_runtime(args: &MayastorCliArgs) {
...
+        error!(
+            "insufficient huge pages allocated, needed:  {}  allocated: {}",
+            PAGES_NEEDED, nr_pages
+        );
+        std::process::exit(1)
+    }
+
+    if free_pages < PAGES_NEEDED {
+        error!(
+            "not enough free pages available: {}, free: {}",
```
Comment (jonathan-teh) [inline mayastor/src/bin/mayastor.rs:83]: Wording is inconsistent and maybe a bit confusing, how about `insufficient huge pages free, needed: {}, free: {}`.

**Review by jkryl — APPROVED** [review]

File: mayastor/src/bin/mayastor.rs (hunk 2 above)
Comment (gila) [reply mayastor/src/bin/mayastor.rs:73]: Yes, and now, more cores require huge pages for the full potential. But it is not like N cores require N huge pages by default. There is a minimum, and that's what this is trying to capture.

File: mayastor/src/bin/mayastor.rs [hunk 4]
```diff
@@ -71,19 +71,21 @@ fn hugepage_check() {
     let free_pages: u32 = sysfs::parse_value(hugepage_path, "free_hugepages")
         .expect("failed to read the number of free pages");
     if nr_pages < PAGES_NEEDED {
-        error!(
-            "insufficient huge pages allocated, needed:  {}  allocated: {}",
-            PAGES_NEEDED, nr_pages
-        );
-        std::process::exit(1)
+        error!(?PAGES_NEEDED, ?nr_pages, "insuffucient pages available");
```
Comment (jonathan-teh) [inline mayastor/src/bin/mayastor.rs:74]: New typo: insufficient (also below)

**Review by tiagolobocastro — APPROVED** [review]

**Comment by gila** [comment]
bors merge

**Comment by gila** [comment]
bors merge

**Pull request merged**